sklearn/utils/estimator_checks.py

```diff
@@ -1443,41 +1443,79 @@ def check_supervised_y_2d(name, estimator_orig):
     assert_allclose(y_pred.ravel(), y_pred_2d.ravel())
 
 
-@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+@ignore_warnings
+def check_classifiers_classes_helper(X, y, name, classifier_orig):
+    classes = np.unique(y)
+    classifier = clone(classifier_orig)
+    if name == 'BernoulliNB':
+        X = X > X.mean()
+    set_random_state(classifier)
+    # fit
+    classifier.fit(X, y)
+
+    y_pred = classifier.predict(X)
+
+    if hasattr(classifier, "decision_function"):
+        decision = classifier.decision_function(X)
+        n_samples, n_features = X.shape
+        if len(classes) == 2:
+            assert_equal(decision.shape, (n_samples, ))
+            dec_pred = (decision.ravel() > 0).astype(np.int)
+            assert_array_equal(classifier.classes_[dec_pred], y_pred)
+        elif (len(classes) == 3 and
+              # 1on1 of LibSVM works differently
+              not isinstance(classifier, BaseLibSVM)):
+            assert_equal(decision.shape, (n_samples, len(classes)))
+            decision_y = np.argmax(decision, axis=1).astype(int)
+            assert_array_equal(classifier.classes_[decision_y], y_pred)
+
+    # training set performance
+    if name != "ComplementNB":
+        # This is a pathological data set for ComplementNB.
+        assert_array_equal(np.unique(y), np.unique(y_pred))
+    if np.any(classifier.classes_ != classes):
+        print("Unexpected classes_ attribute for %r: "
```

Comment:
This is not covered by tests. It would be good idea to add a test in test_estimator_checks that covers the different cases in the new tests (i.e. make sure that they fail with appropriate messages when an estimator is bad, as our library plus code coverage naturally checks the good cases).